### PR TITLE
task-54908: added a new push notification when a user receives a message

### PR DESCRIPTION
**Pull request by Amine0008**
Added a new type of push notification informing the user that he has received a new message

### CODE REVIEW

File: extension/src/main/webapp/WEB-INF/conf/chat/notification-configuration.xml [hunk 1]
```diff
@@ -70,6 +69,45 @@
...
+                        <field name="resourceBundleKey">
+                            <string>UINotification.label.ChatMessageReceivedNotificationPlugin</string>
+                        </field>
+                        <field name="order">
+                            <string>12</string>
+                        </field>
+                        <field name="defaultConfig">
+                            <collection type="java.util.ArrayList">
+                                <value>
+                                    <string>Instantly</string>
```
Comment (ahamdi) [inline extension/src/main/webapp/WEB-INF/conf/chat/notification-configuration.xml]: We have to disable sending push notifications by default, users could activate them if they wish.

File: services/src/main/java/org/exoplatform/addons/chat/model/MessageReceivedModel.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	public String toString() {
+		return "MessageReceivedModel [roomId=" + roomId + ", roomName=" + roomName + ", message=" + message
+				+ ", sender=" + sender + ", senderFullName=" + senderFullName + ", receivers=" + receivers + "]";
+	}
+	
+	
+	
+    
+    
+
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/chat/model/MessageReceivedModel.java]: Let's remove extra lines

Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/chat/model/MessageReceivedModel.java]: Let's remove extra lines

File: services/src/main/java/org/exoplatform/addons/chat/utils/NotificationUtils.java [hunk 3]
```diff
@@ -41,6 +48,8 @@ public static final TemplateContext buildTemplateParameters(TemplateProvider tem
         String language = NotificationPluginUtils.getLanguage(notification.getTo());
         TemplateContext templateContext = getTemplateContext(templateProvider, notification, language);
         Boolean isRead = Boolean.valueOf(notification.getValueOwnerParameter(NotificationMessageUtils.READ_PORPERTY.getKey()));
+        templateContext.put("MESSAGE",notification.getValueOwnerParameter("message"));
+        templateContext.put("IS_GROUPE_CHAT",notification.getValueOwnerParameter("isGroupeChat"));
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/chat/utils/NotificationUtils.java]:
```suggestion
        templateContext.put("IS_GROUP_CHAT",notification.getValueOwnerParameter("isGroupChat"));
```
```suggestion
        templateContext.put("IS_GROUPE_CHAT",notification.getValueOwnerParameter("isGroupeChat"));
```

Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/chat/utils/NotificationUtils.java]:
```suggestion
        templateContext.put("IS_GROUP_CHAT",notification.getValueOwnerParameter("isGroupChat"));
```
```suggestion
        templateContext.put("IS_GROUPE_CHAT",notification.getValueOwnerParameter("isGroupeChat"));
```

File: extension/src/main/webapp/WEB-INF/conf/chat/templates/notification/push/ChatPushPlugin.gtmpl [hunk 4]
```diff
@@ -1 +1,19 @@
-<%=_ctx.appRes("exoplatform.chat.MentionPlugin", USER, ROOM_NAME)%>
\ No newline at end of file
+<%
+def notificationDescription = "";
+switch(NOTIF_TYPE) {
+ case "ChatMentionNotificationPlugin":
+   notificationDescription =  _ctx.appRes("exoplatform.chat.MentionPlugin", USER, ROOM_NAME);
+   break;
+ default:
+    if(Boolean.parseBoolean(IS_GROUPE_CHAT))
```
Comment (ahamdi) [inline extension/src/main/webapp/WEB-INF/conf/chat/templates/notification/push/ChatPushPlugin.gtmpl]:
```suggestion
    if(Boolean.parseBoolean(IS_GROUP_CHAT))
```
```suggestion
    if(Boolean.parseBoolean(IS_GROUPE_CHAT))
```

Comment (ahamdi) [inline extension/src/main/webapp/WEB-INF/conf/chat/templates/notification/push/ChatPushPlugin.gtmpl]:
```suggestion
    if(Boolean.parseBoolean(IS_GROUP_CHAT))
```
```suggestion
    if(Boolean.parseBoolean(IS_GROUPE_CHAT))
```

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: extension/src/main/webapp/WEB-INF/conf/chat/templates/notification/push/ChatPushPlugin.gtmpl [hunk 5]
```diff
@@ -1 +1,17 @@
-<%=_ctx.appRes("exoplatform.chat.MentionPlugin", USER, ROOM_NAME)%>
\ No newline at end of file
+<%
+def notificationDescription = "";
+switch(NOTIF_TYPE) {
+ case "ChatMentionNotificationPlugin":
+   notificationDescription =  _ctx.appRes("exoplatform.chat.MentionPlugin", USER, ROOM_NAME);
+   break;
+ default:
+    if(Boolean.parseBoolean(IS_GROUP_CHAT))
```
Comment (ahamdi) [inline extension/src/main/webapp/WEB-INF/conf/chat/templates/notification/push/ChatPushPlugin.gtmpl:8]: Shoulmdn't we add a message like : "You have received a message in room ROOM_NAME" ?

File: extension/src/main/webapp/WEB-INF/conf/chat/templates/notification/push/ChatPushPlugin.gtmpl [hunk 6]
```diff
@@ -1 +1,17 @@
...
+      notificationDescription = "<b>"+ROOM_NAME+"</b><br>"+USER+": "+MESSAGE
+    else
+      notificationDescription = "<b>"+USER+"</b><br>"+MESSAGE
```
Comment (ahamdi) [inline extension/src/main/webapp/WEB-INF/conf/chat/templates/notification/push/ChatPushPlugin.gtmpl:11]: Shouldn't we add a message like : "USER sent you a new message" ?

Comment (hghorbel) [reply extension/src/main/webapp/WEB-INF/conf/chat/templates/notification/push/ChatPushPlugin.gtmpl:11]: The goal is to have the same notification we already have on Chat app in which we display a part of the message and the user name when we have a one to one chat room, and the Chat room label, the user who sent the message and a part of the message. 
Thus, we can use it like you suggest as an MVP and we improve it. wdyt @ahamdi

Comment (ahamdi) [reply extension/src/main/webapp/WEB-INF/conf/chat/templates/notification/push/ChatPushPlugin.gtmpl:11]: OK, let's keep it as is

**Review by ahamdi — APPROVED** [review]

**Pull request merged**